File: topic_5/Haskell/14-deletee.hs
-- Remove a primeira ocorrência de um número numa lista

deletee x [] = []
deletee n (x:xs)
    | n == x = xs
    | otherwise = [x] ++ deletee n xs

main = do
    a <- readLn :: IO Int
    b <- readLn :: IO [Int]
    print $ deletee a b

File: topic_5/Python/14-deletee.py
# Remove a primeira ocorrência de um número numa lista

def deletee(n, xs):
    cont = 0
    lista = []
    for x in xs:
        if x == n:
            cont += 1
        if cont < 1 or cont > 1 or x != n:
            lista.append(x)
    return lista

print(deletee(5, []) == []) #True
print(deletee(1, [1]) == []) #True
print(deletee(4, [1,3,4]) == [1,3]) #True
print(deletee(3, [4,3,1,3]) == [4,1,3]) #True
print(deletee(5, [1,5,6,9]) == [1,6,9]) #True